### PR TITLE
Ignore entities with no start or end when generating message texts

### PR DESCRIPTION
**Pull request by joejuzl**
**Proposed changes**:
Fixes bug when writing test stories that have entities with no start or end e.g. from `trigger_intent`.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by joejuzl** [comment]
For test stories we don't output the `entities` section. This means that entities with no start and end will just not show up. Do you think we should add an `entities` section to the output?

File: rasa/shared/nlu/training_data/formats/readerwriter.py [hunk 1]
```diff
@@ -85,9 +86,12 @@ def generate_message(message: Dict[Text, Any]) -> Text:
         # format (e.g. `/greet{"name": "Rasa"}) and we don't have to add the NLU
         # entity annotation
         if not text.startswith(INTENT_MESSAGE_PREFIX):
-            entities = sorted(message.get("entities", []), key=lambda k: k["start"])
 
-            for entity in entities:
+            entities = message.get("entities", [])
+            entities_with_start = [e for e in entities if "start" in e]
```
Comment (wochinge) [inline rasa/shared/nlu/training_data/formats/readerwriter.py:91]: how about validating `end` as well?

File: rasa/shared/nlu/training_data/formats/readerwriter.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import json
 from collections import OrderedDict
+from operator import itemgetter
```
Comment (wochinge) [inline rasa/shared/nlu/training_data/formats/readerwriter.py:3]:
```suggestion
import operator
```

File: tests/shared/core/test_events.py [hunk 3]
```diff
@@ -358,6 +377,18 @@ def test_md_format_message_using_long_entity_syntax():
...
+def test_md_format_message_using_long_entity_syntax_no_start_end():
+    formatted = format_message(
+        "I am from Berlin.",
+        intent="location",
+        entities=[
+            {"start": 10, "end": 16, "entity": "city", "value": "Berlin"},
+            {"entity": "country", "value": "Germany", "role": "destination",},
+        ],
+    )
+    assert formatted == """I am from [Berlin](city)."""
```
Comment (wochinge) [inline tests/shared/core/test_events.py:389]:
```suggestion
    assert formatted == "I am from [Berlin](city)."
```

**Review by wochinge — APPROVED** [review]
💯 
Should in my opinion go against 2.4.x

**Review by Imod7 — APPROVED** [review]
🏆 🥇 :)
just a tiny question not directly related to your code but I was just wondering why... From your tests you are calling the `format_message` function and I just checked in the [docs](https://rasa.com/docs/rasa/reference/rasa/shared/core/events#format_message) and I do not understand why quotes appear like that : 
`I am from [Berlin]{&quot;entity&quot;: &quot;city&quot;}.`
which corresponds to [this](https://github.com/RasaHQ/rasa/blob/8258c5500108110525aef16fa20ee0f47f435b1e/rasa/shared/core/events.py#L91) where quotes appear ok.

**Comment by joejuzl** [comment]
>   I just checked in the docs and I do not understand why quotes appear like that : I am from [Berlin]{&quot;entity&quot;: &quot;city&quot;}.

Looks like a bug in the docs - wanna open an issue?

**Comment by Imod7** [comment]
> > I just checked in the docs and I do not understand why quotes appear like that : I am from [Berlin]{"entity": "city"}.
> 
> Looks like a bug in the docs - wanna open an issue?

Done #8288

**Pull request merged**